### PR TITLE
fix: kind installer/Resource not present

### LINKED ISSUE #2794: KinD installer/Resource not present
### Bug description

KinD installer/Resource not present therefore I'm not able to create a kubernetes cluster.

The "KinD Extension" is installed and status runningrnetes cluster with Kind in podman-desktop.

Podman-desktop: 1.0.1 and 1.1.1.0
Podman: 4.5.0 and 4.5.1

image

image

image

### Operating system

Windows 10.0.19045 (22H2) (amd64)

### Version

1.1.0

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #2794

### What does this PR do?

Ensure libraries using NodeJS fetch implementation are proxy compatible. That would include the kind and compose extensions.

**WARNING** This part makes the assumption that the proxy configuration is either httpProxy if the proxy is using HTTP with the browser/Podman Desktop or httpsProxy if using HTTPS (ie configuring both makes no sense). Also this PR does not implement support for NO_PROXY, should be a follow up PR

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

Fixes #2794 

### How to test this PR?

1. Configure your workstation with proxy (no network access except for the proxy)
2. Remove kind or compose extensions
3. Launch Podman Desktop
4. Install kind or compose

### CODE REVIEW

File: packages/main/src/plugin/proxy.ts [hunk 1]
```diff
@@ -131,4 +134,19 @@ export class Proxy {
...
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    globalThis.fetch = function (url: RequestInfo | URL, opts?: any) {
+      const proxyurl = getProxyUrl(_me);
+      if (proxyurl) {
+        opts = Object.assign({}, opts, { dispatcher: new ProxyAgent(proxyurl) });
+      }
+
+      return original(url, opts);
+    };
+  }
```
Comment (benoitf) [inline packages/main/src/plugin/proxy.ts:151]: I'm wondering if it should not go the https://github.com/containers/podman-desktop/blob/main/packages/main/src/plugin/extension-loader.ts#L217-L218

Also I'm wondering about the need of `undici` library

it looks like that for `createHttpPatchedModules` https://www.npmjs.com/package/hpagent is used so it looks like to be a duplicated dependency

Probably some code could be shared with createHttpPatchedModules from proxy-resolver.ts class

**Comment by benoitf** [comment]
cc @dgolovin you invested transparent proxy , there was a missing `fetch` case ?

**Comment by lstocchi** [comment]
Just want to say that i'm setting my env to test this PR. i didn't forget about it and sorry for the super late response.

**Review by lstocchi — COMMENTED** [review]
I'll keep trying but it looks to me that something does not work as expected.

What i've done:
I set up fiddler so i could use it as a proxy. 
I set the proxy to 127.0.0.1:8888 in podman desktop but i receive a http error fetch failed when it tries to fetch kind releases on https://api.github.com/repos/kubernetes-sigs/kind/releases.

If i try the same on postman or on a browser i get the response fine.

**Comment by jeffmaury** [comment]
> I'll keep trying but it looks to me that something does not work as expected.
> 
> What i've done: I set up fiddler so i could use it as a proxy. I set the proxy to 127.0.0.1:8888 in podman desktop but i receive a http error fetch failed when it tries to fetch kind releases on https://api.github.com/repos/kubernetes-sigs/kind/releases.
> 
> If i try the same on postman or on a browser i get the response fine.

Can you see the request going through Fiddler ?

**Comment by lstocchi** [comment]
> > I'll keep trying but it looks to me that something does not work as expected.
> > What i've done: I set up fiddler so i could use it as a proxy. I set the proxy to 127.0.0.1:8888 in podman desktop but i receive a http error fetch failed when it tries to fetch kind releases on https://api.github.com/repos/kubernetes-sigs/kind/releases.
> > If i try the same on postman or on a browser i get the response fine.
> 
> Can you see the request going through Fiddler ?

Yes
image

**Comment by jeffmaury** [comment]
> > > I'll keep trying but it looks to me that something does not work as expected.
> > > What i've done: I set up fiddler so i could use it as a proxy. I set the proxy to 127.0.0.1:8888 in podman desktop but i receive a http error fetch failed when it tries to fetch kind releases on https://api.github.com/repos/kubernetes-sigs/kind/releases.
> > > If i try the same on postman or on a browser i get the response fine.
> > 
> > 
> > Can you see the request going through Fiddler ?
> 
> Yes image

Any idea why it's being rejected ?

**Comment by lstocchi** [comment]
> Any idea why it's being rejected ?

it seems it's a problem with the certificate.
By setting `process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";` it works fine

**Review by lstocchi — APPROVED** [review]
So i guess it's not an actual problem. Just me that i used an internal signed certificate.
LGTM, GJ!! 🚀

**Review by dgolovin — APPROVED** [review]

**Review by benoitf — COMMENTED** [review]
it looks I never sent my review (I have a pending review item), so clicking on submit review even if it's too late

**Pull request merged**